app/src/main/java/com/example/android/mobilecourse/RobotDetails.java

```diff
@@ -53,9 +52,83 @@ private void getIncomingIntent() {
             String imageUrlInfo = getIntent().getStringExtra("image");
 
             setInfo(robotNameInfo, robotCodeInfo, addressInfo, timeInfo, trajectoryInfo, imageUrlInfo);
+        } else if (getIntent().hasExtra("notification_robot_code") &&
+                getIntent().hasExtra("notification_message")) {
+            String robotCode = getIntent().getStringExtra("notification_robot_code");
+            String message = getIntent().getStringExtra("notification_message");
+            showInfoFromNotification(robotCode, message);
         }
     }
 
+    private void showInfoFromNotification(String robotCode, String message) {
+        loadRobotsAndShowWithCode(robotCode);
+        showMessage(message);
+    }
+
+    private void loadRobotsAndShowWithCode(String robotCode) {
```

Comment:
It would be better to separate this method to these: "List <Robot> loadRobots()" and "void displayRobotByCode(String robotCode)". Single method will do certain action.